Adding support for named components and configuration to test framework.

smawson opened the pull request:
Each component can be created with a name and optionally a configuration. This allows multiple echo instances, policy backends, envoy proxies etcetera to be managed independently.

istio-testing commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/istio/istio/pull/11688#" title="Author self-approved">smawson</a>*
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **ozevren**

If they are not already assigned, you can assign the PR to them by writing `/assign @ozevren` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[pkg/OWNERS](https://github.com/istio/istio/blob/master/pkg/OWNERS)**
- **[tests/integration2/OWNERS](https://github.com/istio/istio/blob/master/tests/integration2/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["ozevren"]} -->

smawson commented:
Moved this PR over to master and dropped the other, in theory this is in good state but my git foo is weak so I possibly horribly screwed something up.

istio-testing commented:
@smawson: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-pilot-multicluster-e2e.sh | c114ba0361e62ac763d8f966df55af2dfe716891 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/11688/istio-pilot-multicluster-e2e/8469) | `/test istio-pilot-multicluster-e2e`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@smawson: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-pilot-multicluster-e2e.sh | c114ba0361e62ac763d8f966df55af2dfe716891 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/11688/istio-pilot-multicluster-e2e/8469) | `/test istio-pilot-multicluster-e2e`
prow/istio-integ-k8s-tests.sh | c114ba0361e62ac763d8f966df55af2dfe716891 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/11688/istio-integ-k8s-tests/3473) | `/test istio-integ-k8s-tests`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

smawson commented:
/test istio-pilot-multicluster-e2e
/test istio-integ-k8s-tests

istio-testing commented:
@smawson: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-integ-k8s-tests.sh | c114ba0361e62ac763d8f966df55af2dfe716891 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/11688/istio-integ-k8s-tests/3473) | `/test istio-integ-k8s-tests`
prow/istio-pilot-multicluster-e2e.sh | c114ba0361e62ac763d8f966df55af2dfe716891 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/11688/istio-pilot-multicluster-e2e/8503) | `/test istio-pilot-multicluster-e2e`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/istio/istio/pull/11688#pullrequestreview-203301597" title="Approved">nmittler</a>*, *<a href="https://github.com/istio/istio/pull/11688#" title="Author self-approved">smawson</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[pkg/OWNERS](https://github.com/istio/istio/blob/master/pkg/OWNERS)~~ [nmittler]
- ~~[tests/integration2/OWNERS](https://github.com/istio/istio/blob/master/tests/integration2/OWNERS)~~ [nmittler]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":[]} -->

istio-testing commented:
@smawson: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-pilot-multicluster-e2e.sh | c114ba0361e62ac763d8f966df55af2dfe716891 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/11688/istio-pilot-multicluster-e2e/8530) | `/test istio-pilot-multicluster-e2e`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->
